diag_manager2/fms_diag_object.F90

```diff
@@ -378,6 +424,55 @@ function diag_obj_get_varname(obj) result (rslt)
     rslt = obj%varname
 end function diag_obj_get_varname
 
+subroutine diag_obj_set_varname(obj, the_name)
```

Comment:
Like many other routines, this needs documentation on the subroutine and the variables.